### DISCUSSION TITLE
Return type in closure?

### DISCUSSION THREAD

**Discussion by flavio-paqt** [PHP-CS-Fixer/PHP-CS-Fixer#6865] [Q&A] [answered]
## Support question

I have question, is there a rule I can set so an closures have return type. Now I have to set return types by my self. I am curious if there is a rule that I can set. If not, will this be implemented? 

```
$this->partialMock(
            KibanaClient::class,
            function (MockInterface $interface): void {
                /** @var Expectation $expectation */
                $expectation = $this->addExpectation($interface, 'send');

                $expectation
                    ->once()
                    ->andReturn([
                        'KEY' => 'VALUE',
                    ]);
            }
        );
```

**Comment by Wirone** [comment]
From what you provided as an example it's not clear what is the _current_ and _expected_ behavior. Are you talking about `void`? Please make it more obvious, so it's easier to answer your question 🙂.

**Reply by flavio-paqt** [reply]
Yes that is what I mean.

**Reply by Wirone** [reply, marked as answer]
There's [`void_return`](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/rules/function_notation/void_return.rst) fixer that you might check, but as the name suggests it will only handle `void` return type. I believe we don't have fixer that would infer return type from the function's body and would use it as return type in the signature. Feel free to create feature request, but please have in mind that core team not necessarily will work on it because of other ongoing works.
